test/set.jl

```diff
@@ -0,0 +1,120 @@
+    L_grid = transform(L, model.spectral_transform)
+    
+    L2 = rand(LowerTriangularArray{ComplexF32}, trunc-5, trunc-6, nlayers)    # smaller  
+    L2_trunc = spectral_truncation(L2, size(L, 1, ZeroBased, as=Matrix), size(L, 2, ZeroBased, as=Matrix))
+    L3 = rand(LowerTriangularArray{ComplexF32}, trunc+6, trunc+5, nlayers)    # bigger 
+    L3_trunc = spectral_truncation(L3, size(L, 1, ZeroBased, as=Matrix), size(L, 2, ZeroBased, as=Matrix))
+    
+    A = rand(spectral_grid.Grid{Float32}, spectral_grid.nlat_half, nlayers)       # same grid 
+    A_spec = transform(A, model.spectral_transform)
+    B = rand(OctaHEALPixGrid{Float32}, spectral_grid.nlat_half, nlayers)          # different grid 
+    
+    f(lon, lat, sig) = sind(lon)*cosd(lat)*(1 - sig)
+
+    prog_new = simulation.prognostic_variables
+    prog_old = deepcopy(prog_new)
+
+    # set things ...
+
+    # LTA
+    set!(simulation, vor=L, lf = lf)
+    @test prog_new.vor[lf] == L
+
+    set!(simulation, div=L, lf = lf; add=true)
+    @test prog_new.div[lf] == (prog_old.div[lf] .+ L)
+
+    set!(simulation, temp=L2, lf=lf)
+    @test prog_new.temp[lf] ≈ L2_trunc
+
+    set!(simulation, humid=L3, lf=lf)
+    @test prog_new.humid[lf] ≈ L3_trunc
+    
+    set!(simulation, pres=L[:,1], lf=lf)
+    @test prog_new.pres[lf] == L[:,1]
+
+    set!(simulation, vor=A, lf=lf)
+    @test prog_new.vor[lf] == A_spec
+
+    set!(simulation, vor=A, lf=lf, add=true)
+    @test prog_new.vor[lf] == (2 .* A_spec)
+
+    # grids 
+    set!(simulation, sea_surface_temperature=A[:,1], lf=lf)
+    @test prog_new.ocean.sea_surface_temperature == A[:,1]
+
+    set!(simulation, sea_ice_concentration=B[:,1], lf=lf, add=true)
+    C = similar(A[:,1])
+    SpeedyWeather.RingGrids.interpolate!(C, B[:,1])
+    @test prog_new.ocean.sea_ice_concentration ≈ (prog_old.ocean.sea_ice_concentration .+ C)
+
+    set!(simulation, land_surface_temperature=L[:,1], lf=lf)
+    @test prog_new.land.land_surface_temperature ≈ L_grid[:,1]
+
+    set!(simulation, soil_moisture_layer2=L[:,1], lf=lf)   
+    @test prog_new.land.soil_moisture_layer2 ≈ L_grid[:,1]
+
+    # numbers
+    set!(simulation, vor=Float32(3.), lf=lf)
+    M3 = zeros(spectral_grid.Grid{Float32}, spectral_grid.nlat_half, nlayers) .+ 3  # same grid 
+    M3_spec = transform(M3, model.spectral_transform)
+    @test prog_new.vor[lf] ≈ M3_spec
+
+    set!(simulation, vor=Float32(3.), lf=lf, add=true)
+    @test prog_new.vor[lf] ≈ (2 .* M3_spec)
+
+    set!(simulation, sea_surface_temperature=Float16(3.))
+    @test all(prog_new.ocean.sea_surface_temperature .≈ 3.)
+
+    set!(simulation, sea_surface_temperature=Float16(3.), add=true)
+    @test all(prog_new.ocean.sea_surface_temperature .≈ 6.)
+
+    # vor_div, create u,v first in spectral space
+    u = randn(spectral_grid.SpectralVariable3D, trunc+2, trunc+1, nlayers)
+    v = randn(spectral_grid.SpectralVariable3D, trunc+2, trunc+1, nlayers)
```

Comment:
For any transform tests I usually start in spectral space because the grid space has more degrees of freedom meaning that `rand` of some grid contains a lot of information (high resolution noise) that's lost in the first transform as not even the highest wavenumbers can hold that. You can probably circumvent this by using a linear trunction (instead of the default quadratic) but because we always do quadratic, I'd just make sure to start any spec-grid uv-vor/div roundtrip with information that's surely represented in any space